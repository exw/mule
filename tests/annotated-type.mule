(fn x. x) : (y : {}) -> y
