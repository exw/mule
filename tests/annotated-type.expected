
Desugared:

	(App
	 ((WithType
	   (Fn
	    (()
	     (Annotated
	      (() y (Record (r_info ()) (r_types (() () ())) (r_values (() () ())))))
	     (Var (() y)))))
	  (Lam (x (Var x)))))

inferred type:

	(Fn
	 (() (Record (r_info ()) (r_types (() () ())) (r_values (() () ())))
	  (Record (r_info ()) (r_types (() () ())) (r_values (() () ())))))

Runtime term:

	(Lam (0 () (Var 0)))

Evaluated:

	(Lam (0 () (Var 0)))
